adapters/kidoz/kidoz_test.go

```diff
@@ -0,0 +1,111 @@
+package kidoz
+
+import (
+	"math"
+	"net/http"
+	"testing"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/adapters/adapterstest"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestJsonSamples(t *testing.T) {
+	adapterstest.RunJSONBidderTest(t, "kidoztest", NewKidozBidder("http://example.com/prebid"))
+}
+
+func makeBidRequest() *openrtb.BidRequest {
+	request := &openrtb.BidRequest{
+		ID: "test-req-id-0",
+		Imp: []openrtb.Imp{
+			{
+				ID: "test-imp-id-0",
+				Banner: &openrtb.Banner{
+					Format: []openrtb.Format{
+						{
+							W: 320,
+							H: 50,
+						},
+					},
+				},
+				Ext: []byte(`{"bidder":{"access_token":"token-0","publisher_id":"pub-0"}}`),
+			},
+		},
+	}
+	return request
+}
+
+func TestMakeRequests(t *testing.T) {
+	kidoz := NewKidozBidder("http://example.com/prebid")
+
+	t.Run("Handles Request marshal failure", func(t *testing.T) {
+		request := makeBidRequest()
+		request.Imp[0].BidFloor = math.Inf(1) // cant be marshalled
+		extra := &adapters.ExtraRequestInfo{}
+		reqs, errs := kidoz.MakeRequests(request, extra)
+		// cant assert message its different on different versions of go
+		assert.Equal(t, 1, len(errs))
+		assert.Equal(t, 0, len(reqs))
+	})
+}
+
+func TestMakeBids(t *testing.T) {
+	kidoz := NewKidozBidder("http://example.com/prebid")
+
+	t.Run("Handles response marshal failure", func(t *testing.T) {
+		request := makeBidRequest()
+		requestData := &adapters.RequestData{}
+		responseData := &adapters.ResponseData{
+			StatusCode: http.StatusOK,
+		}
+
+		resp, errs := kidoz.MakeBids(request, requestData, responseData)
+		// cant assert message its different on different versions of go
```

Comment:
If you'd like to perform this test in json, we have a regex matcher to solve the problem between go < 1.13 and 1.13+.

Reply:
I think this tests the important thing, ie that marshalling errors are handled correctly.

Reply:
That's fine. The json tests are not a requirement. It's a bit difficult to understand which error you're expecting here could you do an assert.Contains() for the non-different parts?